manifests/install.pp

```diff
@@ -9,8 +9,16 @@
   $bin_file = "${caddy::install_path}/caddy"
 
   case $caddy::install_method {
-    'repo': { contain caddy::install::repo }
+    'repo': {
+      if $caddy::version {
```

Comment:
In theory, we can set $package_ensure to be $version in this case.. but I'm not sure if that makes any sense 🤔